trin-core/src/types/block_body.rs

```diff
@@ -222,6 +334,75 @@ pub struct EIP1559Transaction {
     pub s: U256,
 }
 
+#[derive(Eq, Debug, Clone, PartialEq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct EIP1559TransactionHelper {
```

Comment:
If I understand the reasoning correctly, the only point for those helper types is to deserialize a transaction?

If this is the case, I think it will be more clear if we add serde's Serialize/Deserialise derive macros on the transaction enum and we can use `#[serde(tag = "type")]` config on the enum. This will convert the enum type to a `type` json field and if we want to rename the fields we can add `#[serde(rename = "0x0")]` (example [here](https://github.com/ogenev/ethportal-api/blob/9b060df420338f1c8aad28500e77a3b13d3404d5/src/types/block_body.rs#L195) ).

This will also require adding Serialize/Deserialize to all enum types, but this is trivial because all ethereum types we use implement them.

That way we will get rid of all helper structs and `Into` traits.